### ISSUE TITLE
Cannot check for Updates - Exception during a connection to https://dbeaver.io/product/version.xml

### ISSUE THREAD

**Issue by johnny68** [dbeaver/dbeaver#35794]
### Description

Cannot Check for updates. 
I manually uninstalled a previous version and installed the current version via brew `brew install --cask dbeaver-community`.

Debug Log: 
```
2024-10-03 11:13:12.490 - DBeaver 24.2.1.202409221725 is starting
2024-10-03 11:13:12.491 - OS: Mac OS X 14.6.1 (aarch64)
2024-10-03 11:13:12.491 - Java version: 17.0.6 by Eclipse Adoptium (64bit)
2024-10-03 11:13:12.491 - Install path: '/Applications/DBeaver.app/Contents/Eclipse'
2024-10-03 11:13:12.491 - Instance path: 'file:/Users/user_name/Library/DBeaverData/workspace6/'
2024-10-03 11:13:12.492 - Memory available 120Mb/1024Mb
2024-10-03 11:13:12.494 - Initialize desktop platform...
2024-10-03 11:13:12.531 - BounceCastle bundle found. Use JCE provider BC
> Start Core File Systems [org.eclipse.core.filesystem 1.11.0.v20240824-0952]
> Start Core Resource Management [org.eclipse.core.resources 3.21.0.v20240805-1607]
> Start Team Support Core [org.eclipse.team.core 3.10.500.v20240621-0541]
> Start DBeaver UI Navigator [org.jkiss.dbeaver.ui.navigator 1.0.139.202409221725]
2024-10-03 11:13:12.679 - Initialize base platform...
> Start DBeaver UI Editors - Base [org.jkiss.dbeaver.ui.editors.base 1.0.139.202409221725]
2024-10-03 11:13:12.687 - Platform initialized (193ms)
2024-10-03 11:13:12.704 - Starting instance server at http://localhost:63514
2024-10-03 11:13:12.706 - Run workbench
2024-10-03 11:13:12.706 - Create display
> Start Eclipse e4 Workbench SWT [org.eclipse.e4.ui.workbench.swt 0.17.500.v20240807-0911]
> Start Eclipse Application Services [org.eclipse.e4.core.services 2.4.400.v20240413-1529]
> Start Eclipse UI Application Services [org.eclipse.e4.ui.services 1.6.300.v20231201-1637]
> Start Eclipse e4 Progress View [org.eclipse.e4.ui.progress 0.4.600.v20240606-1020]
> Start Eclipse Bindings Support [org.eclipse.e4.ui.bindings 0.14.400.v20240321-1245]
> Start Eclipse e4 core commands [org.eclipse.e4.core.commands 1.1.400.v20240424-0444]
> Start EMF XML/XMI Persistence [org.eclipse.emf.ecore.xmi 2.38.0.v20240721-0634]
> Start Eclipse e4 Workbench Add-ons [org.eclipse.e4.ui.workbench.addons.swt 1.5.500.v20240620-1945]
> Start Eclipse e4 Workbench SWT Renderer [org.eclipse.e4.ui.workbench.renderers.swt 0.16.500.v20240727-1037]
> Start Eclipse Dependency Injection Extensions [org.eclipse.e4.core.di.extensions 0.18.300.v20240413-1529]
> Start Eclipse CSS SWT Support [org.eclipse.e4.ui.css.swt 0.15.400.v20240321-1245]
> Start Eclipse CSS SWT Theme Support [org.eclipse.e4.ui.css.swt.theme 0.14.400.v20240424-0956]
> Start Eclipse SDK Themes [org.eclipse.ui.themes 1.2.2500.v20240819-1245]
> Start Default Text Editor [org.eclipse.ui.editors 3.18.0.v20240807-0735]
> Start Text Editor Framework [org.eclipse.ui.workbench.texteditor 3.18.0.v20240819-1419]
2024-10-03 11:13:13.458 - Configure workbench window
> Start DBeaver SQL Model [org.jkiss.dbeaver.model.sql 1.0.129.202409221725]
2024-10-03 11:13:13.507 - Total database drivers: 119 (119)
> Start Help System Core [org.eclipse.help 3.10.400.v20240415-0528]
2024-10-03 11:13:13.539 - Create actions
2024-10-03 11:13:13.542 - Disable Eclipse action set 'org.eclipse.ui.edit.text.actionSet.annotationNavigation'
2024-10-03 11:13:13.542 - Disable Eclipse action set 'org.eclipse.ui.WorkingSetActionSet'
2024-10-03 11:13:13.542 - Disable Eclipse action set 'org.eclipse.ui.actionSet.openFiles'
2024-10-03 11:13:13.542 - Create workbench actions
2024-10-03 11:13:13.727 - Initialize workbench window
2024-10-03 11:13:13.729 - Finish initialization
> Start DBeaver UI Editors - Data [org.jkiss.dbeaver.ui.editors.data 1.0.139.202409221725]
> Start Internet Connection Management UI [org.eclipse.ui.net 1.5.400.v20240413-1529]
> Start Internet Connection Management [org.eclipse.core.net 1.5.500.v20240625-1706]
> Start DBeaver UI Editors - SQL [org.jkiss.dbeaver.ui.editors.sql 1.0.139.202409221725]
> Start DBeaver Sample Database [org.jkiss.dbeaver.ext.sample.database 1.0.176.202409221725]
> Start DBeaver Tip of the day [org.jkiss.dbeaver.ext.ui.tipoftheday 1.0.145.202409221725]
> Start GEF Classic Draw2d [org.eclipse.draw2d 3.17.0.202409021815]
> Start GEF Classic GEF (MVC) [org.eclipse.gef 3.19.0.202409021815]
> Start DBeaver ERD UI [org.jkiss.dbeaver.erd.ui 3.0.99.202409221725]
> Start DBeaver ERD Model [org.jkiss.dbeaver.erd.model 1.0.99.202409221725]
> Start DBeaver Dashboard UI [org.jkiss.dbeaver.ui.dashboard 1.0.136.202409221725]
> Start DBeaver Dashboards Model [org.jkiss.dbeaver.model.dashboard 1.0.12.202409221725]
2024-10-03 11:13:25.232 - Open [https://dbeaver.io/product/version.xml]
2024-10-03 11:13:25.755 - unable to find valid certification path to requested target
sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
	at java.base/sun.security.provider.certpath.SunCertPathBuilder.build(Unknown Source)
	at java.base/sun.security.provider.certpath.SunCertPathBuilder.engineBuild(Unknown Source)
	at java.base/java.security.cert.CertPathBuilder.build(Unknown Source)
	at java.base/sun.security.validator.PKIXValidator.doBuild(Unknown Source)
	at java.base/sun.security.validator.PKIXValidator.engineValidate(Unknown Source)
	at java.base/sun.security.validator.Validator.validate(Unknown Source)
	at java.base/sun.security.ssl.X509TrustManagerImpl.checkTrusted(Unknown Source)
	at java.base/sun.security.ssl.X509TrustManagerImpl.checkServerTrusted(Unknown Source)
	at java.base/sun.security.ssl.CertificateMessage$T12CertificateConsumer.checkServerCerts(Unknown Source)
	at java.base/sun.security.ssl.CertificateMessage$T12CertificateConsumer.onCertificate(Unknown Source)
	at java.base/sun.security.ssl.CertificateMessage$T12CertificateConsumer.consume(Unknown Source)
	at java.base/sun.security.ssl.SSLHandshake.consume(Unknown Source)
	at java.base/sun.security.ssl.HandshakeContext.dispatch(Unknown Source)
	at java.base/sun.security.ssl.HandshakeContext.dispatch(Unknown Source)
	at java.base/sun.security.ssl.TransportContext.dispatch(Unknown Source)
	at java.base/sun.security.ssl.SSLTransport.decode(Unknown Source)
	at java.base/sun.security.ssl.SSLSocketImpl.decode(Unknown Source)
	at java.base/sun.security.ssl.SSLSocketImpl.readHandshakeRecord(Unknown Source)
	at java.base/sun.security.ssl.SSLSocketImpl.startHandshake(Unknown Source)
	at java.base/sun.security.ssl.SSLSocketImpl.startHandshake(Unknown Source)
	at java.base/sun.net.www.protocol.https.HttpsClient.afterConnect(Unknown Source)
	at java.base/sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.connect(Unknown Source)
	at java.base/sun.net.www.protocol.https.HttpsURLConnectionImpl.connect(Unknown Source)
	at org.jkiss.dbeaver.runtime.WebUtils.openURLConnection(WebUtils.java:156)
	at org.jkiss.dbeaver.runtime.WebUtils.openURLConnection(WebUtils.java:87)
	at org.jkiss.dbeaver.runtime.WebUtils.openURLConnection(WebUtils.java:76)
	at org.jkiss.dbeaver.runtime.WebUtils.openConnection(WebUtils.java:58)
	at org.jkiss.dbeaver.runtime.WebUtils.openConnection(WebUtils.java:53)
	at org.jkiss.dbeaver.registry.updater.VersionDescriptor.<init>(VersionDescriptor.java:54)
	at org.jkiss.dbeaver.ui.app.standalone.update.DBeaverVersionChecker.run(DBeaverVersionChecker.java:126)
	at org.jkiss.dbeaver.model.runtime.AbstractJob.run(AbstractJob.java:117)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)
2024-10-03 11:14:08.788 - Open [https://dbeaver.io/product/version.xml]
2024-10-03 11:14:09.206 - unable to find valid certification path to requested target
sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
	at java.base/sun.security.provider.certpath.SunCertPathBuilder.build(Unknown Source)
	at java.base/sun.security.provider.certpath.SunCertPathBuilder.engineBuild(Unknown Source)
	at java.base/java.security.cert.CertPathBuilder.build(Unknown Source)
	at java.base/sun.security.validator.PKIXValidator.doBuild(Unknown Source)
	at java.base/sun.security.validator.PKIXValidator.engineValidate(Unknown Source)
	at java.base/sun.security.validator.Validator.validate(Unknown Source)
	at java.base/sun.security.ssl.X509TrustManagerImpl.checkTrusted(Unknown Source)
	at java.base/sun.security.ssl.X509TrustManagerImpl.checkServerTrusted(Unknown Source)
	at java.base/sun.security.ssl.CertificateMessage$T12CertificateConsumer.checkServerCerts(Unknown Source)
	at java.base/sun.security.ssl.CertificateMessage$T12CertificateConsumer.onCertificate(Unknown Source)
	at java.base/sun.security.ssl.CertificateMessage$T12CertificateConsumer.consume(Unknown Source)
	at java.base/sun.security.ssl.SSLHandshake.consume(Unknown Source)
	at java.base/sun.security.ssl.HandshakeContext.dispatch(Unknown Source)
	at java.base/sun.security.ssl.HandshakeContext.dispatch(Unknown Source)
	at java.base/sun.security.ssl.TransportContext.dispatch(Unknown Source)
	at java.base/sun.security.ssl.SSLTransport.decode(Unknown Source)
	at java.base/sun.security.ssl.SSLSocketImpl.decode(Unknown Source)
	at java.base/sun.security.ssl.SSLSocketImpl.readHandshakeRecord(Unknown Source)
	at java.base/sun.security.ssl.SSLSocketImpl.startHandshake(Unknown Source)
	at java.base/sun.security.ssl.SSLSocketImpl.startHandshake(Unknown Source)
	at java.base/sun.net.www.protocol.https.HttpsClient.afterConnect(Unknown Source)
	at java.base/sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.connect(Unknown Source)
	at java.base/sun.net.www.protocol.https.HttpsURLConnectionImpl.connect(Unknown Source)
	at org.jkiss.dbeaver.runtime.WebUtils.openURLConnection(WebUtils.java:156)
	at org.jkiss.dbeaver.runtime.WebUtils.openURLConnection(WebUtils.java:87)
	at org.jkiss.dbeaver.runtime.WebUtils.openURLConnection(WebUtils.java:76)
	at org.jkiss.dbeaver.runtime.WebUtils.openConnection(WebUtils.java:58)
	at org.jkiss.dbeaver.runtime.WebUtils.openConnection(WebUtils.java:53)
	at org.jkiss.dbeaver.registry.updater.VersionDescriptor.<init>(VersionDescriptor.java:54)
	at org.jkiss.dbeaver.ui.app.standalone.update.DBeaverVersionChecker.run(DBeaverVersionChecker.java:126)
	at org.jkiss.dbeaver.model.runtime.AbstractJob.run(AbstractJob.java:117)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)
2024-10-03 11:19:43.990 - Open [https://dbeaver.io/product/version.xml]
2024-10-03 11:19:44.418 - unable to find valid certification path to requested target
sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
	at java.base/sun.security.provider.certpath.SunCertPathBuilder.build(Unknown Source)
	at java.base/sun.security.provider.certpath.SunCertPathBuilder.engineBuild(Unknown Source)
	at java.base/java.security.cert.CertPathBuilder.build(Unknown Source)
	at java.base/sun.security.validator.PKIXValidator.doBuild(Unknown Source)
	at java.base/sun.security.validator.PKIXValidator.engineValidate(Unknown Source)
	at java.base/sun.security.validator.Validator.validate(Unknown Source)
	at java.base/sun.security.ssl.X509TrustManagerImpl.checkTrusted(Unknown Source)
	at java.base/sun.security.ssl.X509TrustManagerImpl.checkServerTrusted(Unknown Source)
	at java.base/sun.security.ssl.CertificateMessage$T12CertificateConsumer.checkServerCerts(Unknown Source)
	at java.base/sun.security.ssl.CertificateMessage$T12CertificateConsumer.onCertificate(Unknown Source)
	at java.base/sun.security.ssl.CertificateMessage$T12CertificateConsumer.consume(Unknown Source)
	at java.base/sun.security.ssl.SSLHandshake.consume(Unknown Source)
	at java.base/sun.security.ssl.HandshakeContext.dispatch(Unknown Source)
	at java.base/sun.security.ssl.HandshakeContext.dispatch(Unknown Source)
	at java.base/sun.security.ssl.TransportContext.dispatch(Unknown Source)
	at java.base/sun.security.ssl.SSLTransport.decode(Unknown Source)
	at java.base/sun.security.ssl.SSLSocketImpl.decode(Unknown Source)
	at java.base/sun.security.ssl.SSLSocketImpl.readHandshakeRecord(Unknown Source)
	at java.base/sun.security.ssl.SSLSocketImpl.startHandshake(Unknown Source)
	at java.base/sun.security.ssl.SSLSocketImpl.startHandshake(Unknown Source)
	at java.base/sun.net.www.protocol.https.HttpsClient.afterConnect(Unknown Source)
	at java.base/sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.connect(Unknown Source)
	at java.base/sun.net.www.protocol.https.HttpsURLConnectionImpl.connect(Unknown Source)
	at org.jkiss.dbeaver.runtime.WebUtils.openURLConnection(WebUtils.java:156)
	at org.jkiss.dbeaver.runtime.WebUtils.openURLConnection(WebUtils.java:87)
	at org.jkiss.dbeaver.runtime.WebUtils.openURLConnection(WebUtils.java:76)
	at org.jkiss.dbeaver.runtime.WebUtils.openConnection(WebUtils.java:58)
	at org.jkiss.dbeaver.runtime.WebUtils.openConnection(WebUtils.java:53)
	at org.jkiss.dbeaver.registry.updater.VersionDescriptor.<init>(VersionDescriptor.java:54)
	at org.jkiss.dbeaver.ui.app.standalone.update.DBeaverVersionChecker.run(DBeaverVersionChecker.java:126)
	at org.jkiss.dbeaver.model.runtime.AbstractJob.run(AbstractJob.java:117)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)
```

### DBeaver Version

Community 24.2.1.202409221725

### Operating System

Mac OS X 14.6.1

### Database and driver

Not Applicable as I am just checking for updates.

### Steps to reproduce

1. Open up Dbeaver
2. Check for updates for the 'Help' menu item.

### Additional context

I had to recently update the Java Version.
When I check for the Java Version from Terminal the following is the output.
```
java --version
openjdk 17.0.12 2024-07-16 LTS
OpenJDK Runtime Environment Zulu17.52+17-CA (build 17.0.12+7-LTS)
OpenJDK 64-Bit Server VM Zulu17.52+17-CA (build 17.0.12+7-LTS, mixed mode, sharing)
```

I can see there is a difference, i.e.
In the Debug log we have Java version: **17.0.6** and from the terminal output we have **17.0.12** 

The machine I am using is running on M2 Apple Silicon.

**Comment by jramosf** [comment]
I can reproduce this, the issue seems to be misconfiguration on `dbeaver.io` server that is not sending the complete Chain of Trust back to the user:

```
$ openssl s_client -connect dbeaver.io:443
Connecting to 104.131.190.124
CONNECTED(00000005)
depth=0 CN=dbeaver.io
verify error:num=20:unable to get local issuer certificate
verify return:1
depth=0 CN=dbeaver.io
verify error:num=21:unable to verify the first certificate
verify return:1
depth=0 CN=dbeaver.io
verify return:1
---
Certificate chain
 0 s:CN=dbeaver.io
   i:C=GB, ST=Greater Manchester, L=Salford, O=Sectigo Limited, CN=Sectigo RSA Domain Validation Secure Server CA
   a:PKEY: rsaEncryption, 2048 (bit); sigalg: RSA-SHA256
   v:NotBefore: Oct  2 00:00:00 2024 GMT; NotAfter: May  1 23:59:59 2025 GMT
---
Server certificate
-----BEGIN CERTIFICATE-----
MIIGLzCCBRegAwIBAgIRAI1No5lz/4lTOVwmrI6aReUwDQYJKoZIhvcNAQELBQAw
gY8xCzAJBgNVBAYTAkdCMRswGQYDVQQIExJHcmVhdGVyIE1hbmNoZXN0ZXIxEDAO
BgNVBAcTB1NhbGZvcmQxGDAWBgNVBAoTD1NlY3RpZ28gTGltaXRlZDE3MDUGA1UE
AxMuU2VjdGlnbyBSU0EgRG9tYWluIFZhbGlkYXRpb24gU2VjdXJlIFNlcnZlciBD
QTAeFw0yNDEwMDIwMDAwMDBaFw0yNTA1MDEyMzU5NTlaMBUxEzARBgNVBAMTCmRi
ZWF2ZXIuaW8wggEiMA0GCSqGSIb3DQEBAQUAA4IBDwAwggEKAoIBAQCwOLCu3Suq
ahpjeAXkSETvZMozVLZWT3K+f530Ink3mb2pcGNj7BAm917lHCGiJRzXSl2NRYB6
19+rUvxvbr5yVp8HsIluvvCtxN3x9SCoDLMmIWAALMlmYtWNKoasxSrpcXc5B0n5
bMhaZhpxjz9Dytwh+fRIdGr08VxP9h5Au693chI6Z6ysgSeMypzvgDWYBN+PmwhG
/EdwEcAvyeVu5IbWK/7lDYA8itMiEXPC9qC/dejlhoT5h1zA7FSdZlfo3r6gxEgV
0ZX6L8egs2kYM88STrHyoOHuRyZ4NXyxWg5Idu9bVo3yrzHmV4kGq4sh8/IazsgQ
WxMKY+nX6EfbAgMBAAGjggL9MIIC+TAfBgNVHSMEGDAWgBSNjF7EVK2K4Xfpm/mb
BeG4AY1h4TAdBgNVHQ4EFgQUko1b4xLTbUjxfva/c7b6F5ZoF74wDgYDVR0PAQH/
BAQDAgWgMAwGA1UdEwEB/wQCMAAwHQYDVR0lBBYwFAYIKwYBBQUHAwEGCCsGAQUF
BwMCMEkGA1UdIARCMEAwNAYLKwYBBAGyMQECAgcwJTAjBggrBgEFBQcCARYXaHR0
cHM6Ly9zZWN0aWdvLmNvbS9DUFMwCAYGZ4EMAQIBMIGEBggrBgEFBQcBAQR4MHYw
TwYIKwYBBQUHMAKGQ2h0dHA6Ly9jcnQuc2VjdGlnby5jb20vU2VjdGlnb1JTQURv
bWFpblZhbGlkYXRpb25TZWN1cmVTZXJ2ZXJDQS5jcnQwIwYIKwYBBQUHMAGGF2h0
dHA6Ly9vY3NwLnNlY3RpZ28uY29tMCUGA1UdEQQeMByCCmRiZWF2ZXIuaW+CDnd3
dy5kYmVhdmVyLmlvMIIBfwYKKwYBBAHWeQIEAgSCAW8EggFrAWkAdQDPEVbu1S58
r/OHW9lpLpvpGnFnSrAX7KwB0lt3zsw7CAAAAZJMvUOQAAAEAwBGMEQCIBGuuTcF
z10mkP6KIFv5Z56x62xmX4oa2pTyt1oeNURnAiAQxzXYNm8euL3Ik/25v6c/zfgt
+Sf+Kb3W/ygCvTlkwQB3AMz7D2qFcQll/pWbU87psnwi6YVcDZeNtql+VMD+TA2w
AAABkky9Q04AAAQDAEgwRgIhAOFWrNRNk9AiDu9VaA4OBpxsyvSVeSYnXzzy0AT4
ljm+AiEA5osh2sJrK2UHu19ACRS2u0V7B8nuW3QqLcUtgW/OftgAdwBOdaMnXJoQ
wzhbbNTfP1LrHfDgjhuNacCx+mSxYpo53wAAAZJMvUMuAAAEAwBIMEYCIQCP/2Eo
++h27R3QkCKSgUvyXWXxFS24ErBlejdJNn1hMQIhAM8i3Z5jS8VzNpu8VxSFN2QX
dnEsGqZrtV0TjaF019W2MA0GCSqGSIb3DQEBCwUAA4IBAQB9KHjAYyZxzcwDlF1m
fXfgS5qPMdAf6zq1HhprrrYOeGfwdi7K22RUyJD2wiuuG9Vi80s0lvqczkUxPQzn
2GJaeJZ3ejve3eRQPPxsc9pSiQM6fMZ5FODnUx0nUzrXrYLBN0H869U/f0TvUXU5
nZiV+99bRS1/+dKSzborFIQlO/CoOFu45w6wQ7hFLIH3Tale6LX3gfCri68UNQ8Y
Da+4ehxPmDo7EixIoMOVxwu7qwmKYFpnE7u8d/4wTv6x/1nFFcnTCOspMEUIzhuO
whwJWJgFgj4XvVbgPaQXSaovoJeSeuYKRnoMFdXwQurmfGBhErn15cdrvQcqCJy2
GrUs
-----END CERTIFICATE-----
subject=CN=dbeaver.io
issuer=C=GB, ST=Greater Manchester, L=Salford, O=Sectigo Limited, CN=Sectigo RSA Domain Validation Secure Server CA
```


The certificate is signed by the same CA as last year: 
- Issued in 2023: https://crt.sh/?id=10325384562
- Issued in 2024: https://crt.sh/?id=14768846020

❌  The intermediate `i:C=GB, ST=Greater Manchester, L=Salford, O=Sectigo Limited, CN=Sectigo RSA Domain Validation Secure Server CA` is **not** in cacerts in the embedded JRE used by DBeaver, server doesn't send the chain so validation fails.

![image](https://github.com/user-attachments/assets/d3f3b1da-d7f6-4680-92eb-fd37ed887fcb)

**Comment by ghost-from-the-past** [comment]
related to https://github.com/dbeaver/dbeaver/issues/35777

**Comment by E1izabeth** [comment]
Thank you for report

**Comment by E1izabeth** [comment]
@johnny68 @jramosf @ghost-from-the-past 
There was an issue with https certificates on dbeaver.io. Now it should work ok

**Comment by E1izabeth** [comment]
I'm closing it as a resolved and duplicate of #35796 #35777
